### PR TITLE
Add public module curation API docs

### PR DESCRIPTION
**Pull request by joekarl**
## Labels

- [ ] inaccuracy
- [ ] clarification
- [X] new docs
- [ ] cosmetic bug - fixing broken text or markup
- [ ] enhancement - changing the website's behavior/layout
- [X] api - requires an update to the [changelog](https://www.terraform.io/docs/cloud/api/changelog.html)

### CODE REVIEW

File: content/source/docs/cloud/api/modules.html.md [hunk 1]
```diff
@@ -37,13 +37,19 @@ The Terraform Cloud Module Registry implements the [Registry standard API](../..
 - Get a specific module
 - Download the latest version of a module
 
+For publicly curated modules, the Terraform Cloud Module Registry acts as a proxy to the [Terraform Registry](https://registry.terraform.io) for the following:
+- List available versions for a specific module
```
Comment (jbonhag) [inline content/source/docs/cloud/api/modules.html.md:41]: There should be a newline here to start the list:

```suggestion

- List available versions for a specific module
```

File: content/source/docs/cloud/api/modules.html.md [hunk 2]
```diff
@@ -59,11 +65,159 @@ $ curl \
...
 
-Publishes a new registry module from a VCS repository, with module versions managed automatically by the repository's tags. The publishing process will fetch all tags in the source repository that look like [SemVer](https://semver.org/) versions with optional 'v' prefix. For each version, the tag is cloned and the config parsed to populate module details (input and output variables, readme, submodules, etc.). The [Module Registry Requirements](../../registry/modules/publish.html#requirements) define additional requirements on naming, standard module structure and tags for releases.
+Lists the modules that are available to a given organization. This includes the full list of publicly curated and private modules and is filterable.
+
+Status  | Response                                           | Reason
+--------|----------------------------------------------------|----------
+[200][] | [JSON API document][] (`type: "registry-modules"`) | The request was successful
+[404][] | [JSON API error object][]                          | Modules not found or user unauthorized to perform action
+
+### Filtering / Pagination
```
Comment (jbonhag) [inline content/source/docs/cloud/api/modules.html.md:99]: It feels like we should use a table here, like we did with https://github.com/hashicorp/terraform-website/blob/master/content/source/docs/cloud/api/admin/organizations.html.md#query-parameters

**Review by jbonhag — CHANGES_REQUESTED** [review]
Looks good -- however, I think we should switch to a table for filter query params.

We should also include the `403` status code for GET/POST/DELETE operations for registry_name: `public` on Terraform Enterprise.

File: content/source/docs/cloud/api/modules.html.md (hunk 1 above)
Comment (joekarl) [reply content/source/docs/cloud/api/modules.html.md:42]: Fixed

File: content/source/docs/cloud/api/modules.html.md (hunk 2 above)
Comment (joekarl) [reply content/source/docs/cloud/api/modules.html.md:100]: Moved to table

**Comment by bethanyr** [comment]
@joekarl - can you also update the api changelog calling out the new api's we are publishing: https://www.terraform.io/docs/cloud/api/changelog.html

You should be able to follow similar format for other entries - and I guess put June 8 as the release date?

**Review by bethanyr — APPROVED** [review]
LGTM

**Review by jbonhag — APPROVED** [review]
Looks good!

**Pull request merged**